packages/@magic-sdk/react-native/src/react-native-webview-controller.tsx

```diff
@@ -138,7 +139,22 @@ export class ReactNativeWebViewController extends ViewController {
       (event.nativeEvent.url === `${this.endpoint}/send/?params=${encodeURIComponent(this.parameters)}` ||
         event.nativeEvent.url === `${this.endpoint}/send/?params=${this.parameters}`)
     ) {
-      const data: any = JSON.parse(event.nativeEvent.data);
+      // Special parsing logic when dealing with TypedArray in the payload
+      // Such change is required as JSON.stringify will manipulate the object and cause exceptions during parsing
+      // The typed Array is stringified in Mgbox with a flag as notation.
+      const data: any = JSON.parse(event.nativeEvent.data, (key, value) => {
+        try {
+          if (value && typeof value === 'object' && value.flag && value.flag === 'MAGIC_PAYLOAD_FLAG_TYPED_ARRAY') {
+            return new global[value.constructor as keyof Global](value.data.split(','));
+          }
+
+          // silently handles exception and return the original copy
+        } catch (e) {
+          console.log(e);
```

Comment:
Are we logging this error with enough context that it's obvious to the developer where it originates from? If not, we may want to make use of the `MagicSDKWarning` class and log it that way.

Reply:
I had it removed and left empty. The message wasn't quite useful at all to developers as the code block has the original value returned whenever there's an exception. 

If anyone encounters an un-recognizable payload in the future, we may ask the developers to submit their function calls to reproduce the payload.